### DISCUSSION TITLE
Need steps to connect through LAN

### DISCUSSION THREAD

**Discussion by AzranAzwer** [tr3v3r/react-native-esc-pos-printer#178] [Q&A]
I've worked with this plugin for older version with bluetooth. Now I have to use the LAN connection. its hard to find the steps on LAN. please help me on this. (**iOS only**)

**Comment by tr3v3r** [comment]
@AzranAzwer  check that you have the proper setup for your printer.

The Printer IP address should be close to you router IP.

https://www.youtube.com/watch?v=txr7pCdABKA

**Reply by AzranAzwer** [reply]
Thank you @tr3v3r ,
 Yes I have completed til this step.
I need to to know the steps to integrating to my project and keep it connected without stopping.

**Reply by tr3v3r** [reply]
> Thank you @tr3v3r ,
> 
>  Yes I have completed til this step.
> 
> I need to to know the steps to integrating to my project and keep it connected without stopping.

Steps the same for all types of connections. Please check api docs and example folder

**Reply by AzranAzwer** [reply]
hey Thank you for the reply @tr3v3r ,

if all steps are same, how to get the list printers which are connecting through LAN to get connected with application.

In my case there are multiple printers available on LAN and i have to select only 2 of them ?
I used below code and its giving me empty array[]

` const {start, printers} = usePrintersDiscovery()`
`    PrintersDiscovery.start({
      timeout: 2000,
      filterOption: {
        deviceModel: DiscoveryFilterOption.MODEL_ALL,
      },
    })`
 `console.log('printers', printers)`
